### PR TITLE
feat: AI Chat UI

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/6238

### Description (What does it do?)
Adds a component for talking to, and displaying messages from, an AI chat API.

### Screenshots (if appropriate):
Screenshot 2024-12-16 at 1 03 00 PM

### How can this be tested?
1. Make sure you have the repo set it: clone it, `yarn install`, and `yarn start`.
2. View the docs for the AI Chat component at http://localhost:6006/?path=/docs/smoot-design-aichat--docs ... they should make sense.
3. Some other things to check in `AiChat`:
    - the scroll position should stay at bottom of chat window as new messages come in, unless the user has scrolled up
    - markdown is rendered, with links in our red text
    - **Accessibility**: If you have a screenreader (e.g., VoiceOver on MacOS) you could try the screenreader functionality:
        - when a new message is received from the AI, **once it is fully loaded**, it is read aloud.
        - While waiting for responses, you'll hear "loading", "still loading". *This is customizable on the `AiChat` component.*.
4. A few other components were added as building blocks that also have their own stories:
    - ScrollSnap: http://localhost:6006/?path=/docs/smoot-design-scrollsnap--docs
    - SrAnnouncer http://localhost:6006/?path=/docs/smoot-design-screenreaderannouncer--docs
    - VisuallyHidden http://localhost:6006/?path=/docs/smoot-design-visuallyhidden--docs

### Additional Context
The AiChat screenreader functionality is modeled after how https://chatgpt.com/ behaves. It's not ideal—both the implementation here and https://chatgpt.com/ read the raw message content, which is markdown. So you'll hear stuff like "star star" and URLs get red aloud. This should be improved, but this is a decent start.

### CODE REVIEW

File: .storybook/main.ts [hunk 1]
```diff
@@ -18,10 +18,10 @@ const config: StorybookConfig = {
   staticDirs: ["./public"],
 
   addons: [
-    getAbsolutePath("@storybook/addon-links"),
-    getAbsolutePath("@storybook/addon-essentials"),
-    getAbsolutePath("@storybook/addon-interactions"),
-    getAbsolutePath("@storybook/addon-webpack5-compiler-swc"),
```
Comment (ChristopherChudzicki) [inline .storybook/main.ts:24]: works around a bug in Storybook that occurs when you try and run it on Node 22.12.0, detailed here: https://github.com/nodejs/node/issues/56127

**Review by jonkafton — COMMENTED** [review]
Works really well. Good clean chat component. Not much to complain about here! Nice separation of the ScrollSnap and SrAnnouncer (can we rename this ScreenReaderAnnouncer?). Vercel's `ai` package seems a sensible choice (how did they secure that package name?!). Interested to see what server side support for native fetch ReadableStream will look like if we're intending to interface the LLMs via our APIs; is the plan to write a [custom provider](https://sdk.vercel.ai/providers/community-providers/custom-providers)?

Two feature requests (reasonably out of scope for this PR):

- A stop button will be handy (LLMs are chatty!). Often the send message button is replaced by a stop button while the reply is mid stream.

- It would be good to also support starter messages (perhaps renamed suggestions) after each response, ie. the LLM is also prompted for new suggestions based on the most recent messages.

File: src/components/ScrollSnap/ScrollSnap.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+const Scroller = styled.div({
+  overflow: "auto",
+})
+
+type ScrollSnapProps = {
+  /**
+   * Tolerance within which scroll will be considered "at the bottom" of the element.
+   */
+  threshold?: number
```
[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply src/components/ScrollSnap/ScrollSnap.tsx:27]: This should be fixed now. I believe `threshold` was working fine initially, but not quite right if you updated its value.

**Comment by ChristopherChudzicki** [comment]
> Interested to see what server side support for native fetch ReadableStream will look like if we're intending to interface the LLMs via our APIs; is the plan to write a [custom provider](https://sdk.vercel.ai/providers/community-providers/custom-providers)?

As long as our chat API streams regular text, I don't think we'll need one, but worth reading up on.

**Review by jonkafton — APPROVED** [review]

**Pull request merged**